src/components/tasks/TaskWrapper.js

```diff
@@ -1,30 +1,20 @@
-import {CodeEditor} from 'e-stella-code-editor'
-import {tasksApi} from "../../utils/apis/tasksAPI";
-import {useParams} from "react-router-dom";
 import {codeCheckerLink} from "../../utils/apis/APILinks";
+import {CodeEditor} from "e-stella-code-editor";
+import React from "react";
 import {useAbly} from "../../utils/hooks/useAbly";
 
-export const TaskWrapper = ({ id: propId, toSolveTask = true, submitLeftOffset = 0 }) => {
-
-    let { taskStageUUID } = useParams()
-
-    if(!taskStageUUID) taskStageUUID = propId
-
-    const { pub, sub, clientId } = useAbly(`codeChanged/${taskStageUUID}`)
-
-    const fetchTasks = toSolveTask ? () => tasksApi.getTasks(taskStageUUID || NaN) : () => new Promise(_ => {})
+const TaskWrapper = ({task, taskStageUUID, toSolveTask = true, submitLeftOffset = 0}) => {
+    const { pub, sub, clientId } = useAbly(`codeChanged/${taskStageUUID}/${task.id}`)
     const outerMonacoWrapperStyle = toSolveTask ? null : { height: '60vh' }
 
-    return (
-        <div>
-            <CodeEditor
-                codeCheckerBaseLink={codeCheckerLink}
-                fetchTasks={fetchTasks}
-                solverId={taskStageUUID}
-                absoluteOffset={{settings: { top: 3, right: 0 }, submit: {top: 3, left: submitLeftOffset}}}
-                outerMonacoWrapperStyle={outerMonacoWrapperStyle}
-                sharingCodeFunctions={{ pub, sub, id: clientId }}
-             />
-        </div>
-    )
-}
\ No newline at end of file
+    return <CodeEditor
+        codeCheckerBaseLink={codeCheckerLink}
+        fetchTasks={() => new Promise(resolve => resolve([task]))}
+        solverId={taskStageUUID}
+        absoluteOffset={{settings: { top: 0, right: 0 }, submit: {top: 0, left: submitLeftOffset}}}
```

Comment:
Stawiałeś to i oczywiście przyciski są dobrze widoczne zarówno podczas rozmowy jak i standalone zadania, tak?

Reply:
Zapomniałem w sumie zobaczyć podczas rozmowy, jak sprawdzę to odpowiem, ale dla widoku standalone jest git

Reply:
Popieram @Davenury

Reply:
Śmiga tak, że aż się dymi za tą karuzelą